server/game/core/event/EventWindow.js

```diff
@@ -57,6 +59,12 @@ class EventWindow extends BaseStepWithPipeline {
     setCurrentEventWindow() {
         this.previousEventWindow = this.game.currentEventWindow;
         this.game.currentEventWindow = this;
+        if (this.resolveTriggersAfter) {
+            this.triggeredAbilityWindow = new TriggeredAbilityWindow(this.game, this, AbilityType.Triggered);
+        } else if (this.previousEventWindow) {
+            this.triggeredAbilityWindow = this.previousEventWindow.triggeredAbilityWindow;
+            this.triggeredAbilityWindow.addTriggeringEvents(this.events);
```

Comment:
We actually can't set the triggering events here because it's possible that the set of events may change before we reach `executeHandler`. For example, `createContingentEvents` could do it - it is used for generating the events that defeat upgrades if the owning unit is defeated, as well as potentially some additional cases down the road. Replacement effects can also add new events.

Also, I suspect that once we do a deep dive on unit defeat timing we will find that we need to insert defeat events into the window in the middle of resolution (see my comment on the triggered ability window tests). So all that to say, better if we keep the set of events being triggered as flexible as possible until `executeHandler` is totally done.

Reply:
So, do this(line 66) in executeHandler instead? I'm not sure if the TriggeredEventWindow would need to know this EventWindow's events in the meantime.

Reply:
Yeah I think so, iirc that's the way it was beforehand so that should still work